libtransmission/crypto-utils.h

```diff
@@ -70,6 +72,44 @@ std::optional<tr_sha1_digest_t> tr_sha1(T... args)
     return std::nullopt;
 }
 
+/**
+ * @brief Allocate and initialize new SHA1 hasher context.
+ */
+tr_sha256_ctx_t tr_sha256_init(void);
+
+/**
+ * @brief Update SHA1 hash.
+ */
+bool tr_sha256_update(tr_sha256_ctx_t handle, void const* data, size_t data_length);
+
+/**
+ * @brief Finalize and export SHA1 hash, free hasher context.
+ */
+std::optional<tr_sha256_digest_t> tr_sha256_final(tr_sha256_ctx_t handle);
+
+/**
+ * @brief generate a SHA256 hash from some memory
+ */
+template<typename... T>
+std::optional<tr_sha256_digest_t> tr_sha256(T... args)
```

Comment:
```
31][Step 2/4] In file included from /usr/home/user/buildAgent4/work/1f3bff43dd55cfe9/tests/libtransmission/crypto-test.cc:19:
[21:48:31][Step 2/4] In file included from /usr/home/user/buildAgent4/work/1f3bff43dd55cfe9/tests/libtransmission/crypto-test-ref.h:65:
[21:48:31][Step 2/4] In file included from /usr/home/user/buildAgent4/work/1f3bff43dd55cfe9/libtransmission/crypto.h:18:
[21:48:31][Step 2/4] /usr/home/user/buildAgent4/work/1f3bff43dd55cfe9/libtransmission/crypto-utils.h:94:35: error: redefinition of 'tr_sha256'
[21:48:31][Step 2/4] std::optional<tr_sha256_digest_t> tr_sha256(T... args)
[21:48:31][Step 2/4]                                   ^
[21:48:31][Step 2/4] /usr/home/user/buildAgent4/work/1f3bff43dd55cfe9/libtransmission/crypto-utils.h:94:35: note: previous definition is here
[21:48:31][Step 2/4] std::optional<tr_sha256_digest_t> tr_sha256(T... args)
[21:48:31][Step 2/4]                         
```

Reply:
This is probably due to misunderstanding of how the crypto- files work. The design predates the C++ migration and has an odd way of plugging in implementations via `#include`. Totally out-of-scope for this PR but if anyone wanted to make a pure virtual interface class for the crypto API and then have impl .cc filefs for the different providers that would be :100:

Reply:
I don't know how to fix that. @ckerr help?